### PR TITLE
Oceanwater 850 Fix arm check test calling deliver with parameters

### PR DESCRIPTION
**Pull request by AstroStucky**
## Linked Issues:
| EPIC ⚡ | N / A |
| :----------- | :----------- |
| Jira Ticket 🎟️ | [OCEANWATER-850](https://babelfish.arc.nasa.gov/jira/browse/OCEANWATER-850) |
| Github :octocat:  | # |

## Summary of Changes
* arm_check test no longer calls deliver service with parameters
* arm_check now normalizes its joint goals before comparing with the normalized joint states
* arm_check uses arm_deliver_final as the goal for the deliver test

## Test
@mogumbo, feel free to just do a code review, if that's all you've got time for.
@anjan1984, when you test this and we merge it into release, then we can count the results as **Tester 3** on the test page.

### Procedure
Perform [OW-TEST-009-3 Sample Collection](https://babelfish.arc.nasa.gov/confluence/display/OCEANWATERS/OW-TEST-009-3+Sample+Collection) using rostest and ensure all tests pass.
```rostest ow_sim_tests arm_check.test```

### CODE REVIEW

File: ow_sim_tests/test/arm_check.py [hunk 1]
```diff
@@ -89,7 +90,7 @@ def _test_activity(self, service_name, service_type,
       norm_joints_state_array = np.array(
           [self._normalize_angle(e) for e in joints_state])
       joints_abs_diff_array = np.abs(
-          joints_goal_array - norm_joints_state_array)
+          norm_joints_goal_array - norm_joints_state_array)
```
Comment (mogumbo) [inline ow_sim_tests/test/arm_check.py:93]: I think it's safer to normalize to {-pi, pi} after this subtraction happens. If you normalize both values before, there's a small chance your difference could end up being close to 2*pi, much greater than the tolerance.

**Review by mogumbo — CHANGES_REQUESTED** [review]

File: ow_sim_tests/test/arm_check.py (hunk 1 above)
Comment (AstroStucky) [reply ow_sim_tests/test/arm_check.py]: That makes sense. I wanted to prove it to myself with a test. So setting `l_hand_yaw` to exactly pi for `arm_unstowed` in **lander.rsdf**, I did confirm this can happen, and it happens only half the time, which makes sense. I would expect the error depends on whether final joint state is higher/lower than joint goal.
```[rosout][INFO] 2021-11-23 11:32:58,055: 
      current joints state: [ 0.    1.57 -1.57 -0.   -3.14  0.01]
      diff joints state: [0.   0.   0.   0.   6.28 0.01]
      maximum diff between expected/current: 6.28
[rosout][INFO] 2021-11-23 11:32:58,257: operation /arm/unstow took 30s, time-limit is 30s
```
I get the same fails half of the time result when using -pi.

As of commit 3a22af it now normalizes after taking the difference between the angles.

**Comment by AstroStucky** [comment]
Latest commit fixes a normalize angle edge case. Please retest this.

I also changed the test print statement to include negatives in the difference values. So instead of 
```
[rosout][INFO] 2021-11-23 13:02:36,962: 
      current joints state: [ -0.     1.57   4.71  -0.   -11.     0.02]
      diff joints state: [ 0.    0.    0.    0.   0.   0.02]
      maximum diff between expected/current: 0.02
```
it will show
```
[rosout][INFO] 2021-11-23 13:02:36,962: 
      current joints state: [ -0.     1.57   4.71  -0.   -11.     0.02]
      diff joints state: [ 0.    0.    0.    0.   -0.   -0.02]
      maximum diff between expected/current: -0.02
```
It's a subtle change, but I think the latter gives more information, so is therefore a better choice.

**Comment by anjan1984** [comment]
Ran twice and all the tests are successful 
[ROSTEST]-----------------------------------------------------------------------

[ow_sim_tests.rosunit-arm_check/test_01_unstow][passed]
[ow_sim_tests.rosunit-arm_check/test_02_guarded_move][passed]
[ow_sim_tests.rosunit-arm_check/test_03_unstow][passed]
[ow_sim_tests.rosunit-arm_check/test_04_grind][passed]
[ow_sim_tests.rosunit-arm_check/test_05_dig_circular][passed]
[ow_sim_tests.rosunit-arm_check/test_06_grind][passed]
[ow_sim_tests.rosunit-arm_check/test_07_dig_linear][passed]
[ow_sim_tests.rosunit-arm_check/test_08_deliver_sample][passed]
[ow_sim_tests.rosunit-arm_check/test_09_unstow][passed]
[ow_sim_tests.rosunit-arm_check/test_10_stow][passed]

SUMMARY
 * RESULT: SUCCESS
 * TESTS: 10
 * ERRORS: 0
 * FAILURES: 0

**Review by anjan1984 — APPROVED** [review]

**Review by mogumbo — APPROVED** [review]

File: ow_sim_tests/test/arm_check.py (hunk 1 above)
Comment (Samahu) [reply ow_sim_tests/test/arm_check.py]: @mogumbo You comment is right but doesn't apply to the case of the test. We are checking for equal angle values so when both values are normalized and they are equal they wouldn't exceed the {-pi, pi} otherwise they are not equal to start with. I reverted the changes made to this code block to its original implementation to keep it simple.
